gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageImpl.java

```diff
@@ -276,11 +278,15 @@ public GoogleCloudStorageImpl(GoogleCloudStorageOptions options, Storage storage
           httpRequestInitializer instanceof RetryHttpInitializer,
           "request initializer must be an instance of the RetryHttpInitializer class"
               + " when gRPC API enabled");
+      Credential credential = ((RetryHttpInitializer) httpRequestInitializer).getCredential();
+      boolean useDirectpath = credential != null && java.util.Objects.equals(credential.getTokenServerEncodedUrl(),
+          ComputeCredential.TOKEN_SERVER_ENCODED_URL);
```

Comment:
For future reference, I dont think this is the right place for the `useDirectPath` logic , we might be better off with a Factory impl here, so that this logic does not get duplicated now that it is extracted out of the `StorageStubProvider` class

Reply:
Done.